### PR TITLE
fix/file-download-refactor

### PR DESCRIPTION
**Pull request by chesterkmr**
### Description
refactored mapping of persisted files to flow data

### Related issues
 * Provide a link to each related issue.

### Breaking changes
 * Describe the breaking changes that this pull request introduces.

### How these changes were tested
 * Describe the tests that you ran to verify your changes, including devices, operating systems, browsers and versions.

### Examples and references
 * Links, screenshots, and other resources related to this change.

### Checklist
- [] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings and errors
- [] New and existing tests pass locally with my changes

### CODE REVIEW

File: examples/kyb-app/src/hooks/useActiveWorkflowQuery/useActiveWorkflowQuery.ts [hunk 1]
```diff
@@ -1,96 +1,18 @@
...
 import { useQuery } from '@tanstack/react-query';
-import { useEffect, useRef, useState } from 'react';
 
 export const useActiveWorkflowQuery = (documentConfigurations: DocumentConfiguration[]) => {
   const { user } = useSessionQuery();
-  const [isLoading, setLoading] = useState(false);
-  const { data: _flowData, isFetching } = useQuery({
-    ...collectionFlowQuerykeys.getFlowData({ endUserId: user?.id }),
+  const { data: flowData = null, isFetching } = useQuery({
+    ...collectionFlowQuerykeys.getFlowData({ endUserId: user?.id, documentConfigurations }),
```
Comment (Omri-Levy) [inline examples/kyb-app/src/hooks/useActiveWorkflowQuery/useActiveWorkflowQuery.ts:8]: It seems some places use the word `get` and some `fetch`. If it's an async HTTP request it should be `fetch`. :)

File: examples/kyb-app/src/domains/collection-flow/helpers/extract-files-metadata.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+export const extractFilesMetadata = (
+  flowData: FlowData,
+  config: DocumentConfiguration[],
+): WorkflowFileMetadata[] => {
+  const filesMetadata = config.reduce((filesPayload, config) => {
+    const value = traverseObjectAndPickValue(config.name, flowData, null) as unknown;
+
+    if (isFileId(value)) {
```
Comment (Omri-Levy) [inline examples/kyb-app/src/domains/collection-flow/helpers/extract-files-metadata.ts:20]: What was the need for explicitly testing for CUID here versus `undefined` or `string`?

File: examples/kyb-app/src/domains/collection-flow/collection-flow.api.ts [hunk 3]
```diff
@@ -61,7 +65,18 @@ export const fetchActiveWorkflow = async (dto: GetActiveWorkflowDto): Promise<Fl
     })
     .json<{ result: FlowData }>();
 
-  return result.result;
+  const flowData = result.result;
+  const filesMetadata = extractFilesMetadata(flowData, dto.documentConfigurations);
+
+  const files = await downloadAndBuildFiles(filesMetadata);
```
Comment (Omri-Levy) [inline examples/kyb-app/src/domains/collection-flow/collection-flow.api.ts:71]: It seems `kyb-app` has many places which were supposed to use `react-query`. Not sure if an in memory store is needed between the URL, global React state, and `react-query`.

File: examples/kyb-app/src/hooks/useActiveWorkflowQuery/useActiveWorkflowQuery.ts [hunk 4]
```diff
@@ -1,96 +1,18 @@
...
-  }, [documentConfigurations, flowData, isLoading]);
-
-  console.log('FLOW DATA', flowData);
-
   return {
     flowData,
-    isFetching: isFetching || isLoading,
-    workflow: _flowData ? _flowData.workflow : null,
+    isFetching: isFetching,
+    workflow: flowData ? flowData.workflow : null,
```
Comment (Omri-Levy) [inline examples/kyb-app/src/hooks/useActiveWorkflowQuery/useActiveWorkflowQuery.ts:16]: Does it have to be `null` here over `undefined`?

File: examples/kyb-app/src/common/utils/is-cuid.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+export const isCuid = (str: unknown): str is string => {
+  if (typeof str !== 'string') return false;
+
+  const CUIDRegExp = /^[a-z0-9]{7,}$/;
```
Comment (Omri-Levy) [inline examples/kyb-app/src/common/utils/is-cuid.ts]: Note CUIDs have a very specific max length. Usually it's recommended to use the `cuid`'s package `isCuid` util for this. Still unsure if we need to know about the id on the frontend.

File: examples/kyb-app/src/domains/collection-flow/helpers/extract-files-metadata.ts (hunk 2 above)
Comment (chesterkmr) [reply examples/kyb-app/src/domains/collection-flow/helpers/extract-files-metadata.ts:20]: This version inlcudes feature where we can define that document type is an URL.In this case those values should not be processed so in this code im looking only for CUID and using them to download files metadata from /storage

File: examples/kyb-app/src/domains/collection-flow/collection-flow.api.ts (hunk 3 above)
Comment (chesterkmr) [reply examples/kyb-app/src/domains/collection-flow/collection-flow.api.ts:71]: In memory storage is needed to preserve fileIds so we can avoid re-uploads of same files if they are already exists.

File: examples/kyb-app/src/hooks/useActiveWorkflowQuery/useActiveWorkflowQuery.ts (hunk 4 above)
Comment (chesterkmr) [reply examples/kyb-app/src/hooks/useActiveWorkflowQuery/useActiveWorkflowQuery.ts:16]: Assuming any truthy value is Workflow

**Pull request closed without merging**